extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -0,0 +1,417 @@
+    const extensions = pkey as unknown as { raw: string, [internal]: OpenPGP.key.Key };
+    extensions[internal] = pubkey;
+    extensions.raw = raw || pubkey.armor();
+    return pkey;
+  }
+
+  /**
+   * Returns signed data if detached=false, armored
+   * Returns signature if detached=true, armored
+   */
+  public static sign = async (signingPrivate: Key, data: string, detached = false): Promise<string> => {
+    const signingPrv = OpenPGPKey.unwrap(signingPrivate);
+    const message = opgp.cleartext.fromText(data);
+    const signRes = await opgp.sign({ message, armor: true, privateKeys: [signingPrv], detached });
+    if (detached) {
+      if (typeof signRes.signature !== 'string') {
+        throw new Error('signRes.signature unexpectedly not a string when creating detached signature');
+      }
+      return signRes.signature;
+    }
+    return await opgp.stream.readToEnd((signRes as OpenPGP.SignArmorResult).data);
+  }
+
+  public static revoke = async (key: Key): Promise<string | undefined> => {
+    let prv = OpenPGPKey.unwrap(key);
+    if (! await prv.isRevoked()) {
+      prv = await prv.revoke({});
+    }
+    const certificate = await prv.getRevocationCertificate();
+    if (!certificate) {
+      return undefined;
+    } else if (typeof certificate === 'string') {
+      return certificate;
+    } else {
+      return await opgp.stream.readToEnd(certificate);
+    }
+  }
+
+  public static armor = (pubkey: Key): string => {
+    if (pubkey.type !== 'openpgp') {
+      throw new Error('Unsupported key type: ' + pubkey.type);
+    }
+    const extensions = pubkey as unknown as { raw: string };
+    if (!extensions.raw) {
+      throw new Error('Object has type == "openpgp" but no raw key.');
+    }
+    return extensions.raw;
+  }
+
+  private static unwrap = (pubkey: Key) => {
+    if (pubkey.type !== 'openpgp') {
+      throw new Error('Unsupported key type: ' + pubkey.type);
+    }
+    const raw = (pubkey as unknown as { [internal]: OpenPGP.key.Key })[internal];
+    if (!raw) {
+      throw new Error('Object has type == "openpgp" but no internal key.');
+    }
+    return raw;
+  }
+
+  private static usableButExpired = async (key: OpenPGP.key.Key, exp: Date | number | null, expired: () => boolean): Promise<boolean> => {
+    if (!key) {
+      return false;
+    }
+    if (!await Catch.doesReject(key.getEncryptionKey())) {
+      return false;
+    }
+    if (exp === null || typeof exp === 'number') {
+      // If key does not expire (exp == Infinity) the encryption key should be available.
+      return false;
+    }
+    const oneSecondBeforeExpiration = exp && expired() ? new Date(exp.getTime() - 1000) : undefined;
+    if (typeof oneSecondBeforeExpiration === 'undefined') {
+      return false;
+    }
+    try {
+      await key.getEncryptionKey(undefined, oneSecondBeforeExpiration);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+}
+
+export class PgpKey {
```

Comment:
I moved it here to have pgp key things together, we could do something about this in another PR